lib/protocol/constants/errors.js

```diff
@@ -269,8 +333,8 @@ exports.ER_NO_SUCH_TABLE
 exports.ER_NONEXISTING_TABLE_GRANT                                                       = 1147;
 exports.ER_NOT_ALLOWED_COMMAND                                                           = 1148;
 exports.ER_SYNTAX_ERROR                                                                  = 1149;
-exports.ER_DELAYED_CANT_CHANGE_LOCK                                                      = 1150;
-exports.ER_TOO_MANY_DELAYED_THREADS                                                      = 1151;
+exports.OBSOLETE_ER_UNUSED1                                                              = 1150;
```

Comment:
There still seems to be something wrong here, as it seems these are still being changed to say obsolete.

Reply:
That's what it says in 5.7.31! I downloaded 5.7.29 and it's also marked as "UNUSED1". So those are probably from an even older version. Checking the blame here, I see you purposefully omitted these kinds of re-names in the past.

Reply:
hmmm, strange. I don't recall manually omitting changes, but it's certainly possible :)

Reply:
As a check I just re-downloaded the MySQL 5.7.29 source code and ran the current tool against that source tree and there is no changes produced to this file, so something more is going on here 🤔

Reply:
Ah, so this the tool is supposed ignore previously defined codes that are marked as "unused". See the `keepUnusedCodes`, it checks for `ER_UNUSED[0-9]*`, and falls back to what the file had previously. In `fixupCode`, all the `OBSOLETE` markers are supposed to be removed, but it only checks if the string starts with `ER_OBSOLETE_`, not just `OBSOLETE_`. That would be fine since MySQL 5 defines unused codes as `ER_UNUSED[0-9]*`. _However_ my first commit ran the tool with _only_ MySQL 8. MySQL 8 started using `OBSOLETE_ER_UNUSED[0-9]*` instead of `ER_UNUSED[0-9]*`. Subsequent commits perfecting the tool were then using a version of `errors.js` with the older codes already removed.

The solution?

Re-run the new tool with an completely un-altered version of the `errors.js` file from master.